### ISSUE TITLE
[Jamf 10.15] JSSImporter does not detect values of certain fields within an AutoPkg recipe

### ISSUE THREAD

**Issue by wburnett** [jssimporter/JSSImporter#164]
Morning folks. This issue has happened both to me on Autopkg 1.1/JSSimporter 1.0.2b6 and a coworker running AutoPkg v1.04/JSSImporter v1.0.1b4. I updated to the production release of 1.0.2 and this behavior persisted. 

Background - my company is a Jamf MSP that maintains multiple servers for different clients. This behavior is not present on our production servers running 10.13, but is present on the two that we've upgraded to 10.15 for testing.

When running self service recipes with autopkg, JSSImporter does not detect that I have set a location for Self Service icons and descriptions, output snippet below. Some of these are expected, since I haven't set values for all, but I do have a self service description and icon set.

`JSSImporter: No value supplied for self_service_description, setting default value of: 
JSSImporter: No value supplied for package_priority, setting default value of: 10
JSSImporter: No value supplied for JSS_VERIFY_SSL, setting default value of: True
JSSImporter: No value supplied for policy_action_type, setting default value of: Install
JSSImporter: No value supplied for package_info, setting default value of: 
JSSImporter: No value supplied for self_service_icon, setting default value of: 
JSSImporter: No value supplied for force_policy_state, setting default value of: True
JSSImporter: No value supplied for package_boot_volume_required, setting default value of: True
JSSImporter: No value supplied for skip_scope, setting default value of: False
JSSImporter: No value supplied for skip_scripts, setting default value of: False
JSSImporter: No value supplied for os_requirements, setting default value of: 
JSSImporter: No value supplied for package_notes, setting default value of: 
JSSImporter: No value supplied for package_reboot, setting default value of: False
JSSImporter: No value supplied for JSS_SUPPRESS_WARNINGS, setting default value of: True`

Settings for AutoPkg below:

`defaults read com.github.autopkg
{
    "ICON_DIR" = "~/Library/AutoPkg/service-icons";
    "JSS_TEMPLATE_DIR" = "~/Library/AutoPkg/JSS_Templates";
snip
}`

Snippet of recipe:
`	        <key>SELFSERVICE_DESCRIPTION</key>
<string>You can see and find your files in Google Drive on your computer with Drive File Stream. You can save specific files and folders offline, including Team Drives; View and organize your files in Finder on macOS; Open files with apps installed on your computer; See all of your files in Google Drive without using storage space on your computer.</string>
<key>SELFSERVICE_ICON</key>
<string>%ICON_DIR%/GoogleDriveFilestream.png</string>`

The above settings work fine on instances running Jamf 10.13, but break on 10.15. This also occurs when I explicitly declare the path to the service icons folder within the recipe, rather than the interpolated value for ICON_DIR from my settings. Best guess, some piece of the API that JSSImporter is using changed between those versions?

Let me know if I've missed any clarifying details or other necessary information and I'll be happy to help dig further.

**Comment by grahampugh** [comment]
Hi @wburnett can I just make sure that `SELFSERVICE_DESCRIPTION` and `SELFSERVICE_ICON` are consistent through your recipes and Policy Templates? 

They are normally named:
`SELF_SERVICE_DESCRIPTION`
`SELF_SERVICE_ICON`

**Comment by wburnett** [comment]
Ah, good idea. Interesting though, some recipes on older servers are named `SELFSERVICE_DESCRIPTION` and run without issue. 

And it looks like I threw the alarm bell too early - sorry if that gave you a panic. I ran a few other overrides and they're working fine (also with the SELFSERVICE_DESCRIPTION variable, not SELF_SERVICE_DESCRIPTION). 

Unfortunately no dice there though. I changed the variable names to what you suggested in this particular override, and had the same output as above.

`JSSImporter: No value supplied for self_service_description, setting default value of: 
JSSImporter: No value supplied for package_priority, setting default value of: 10
JSSImporter: No value supplied for JSS_VERIFY_SSL, setting default value of: True
JSSImporter: No value supplied for policy_action_type, setting default value of: Install
JSSImporter: No value supplied for package_info, setting default value of: 
JSSImporter: No value supplied for self_service_icon, setting default value of: 
JSSImporter: No value supplied for force_policy_state, setting default value of: True
JSSImporter: No value supplied for package_boot_volume_required, setting default value of: True
JSSImporter: No value supplied for skip_scope, setting default value of: False
JSSImporter: No value supplied for skip_scripts, setting default value of: False
JSSImporter: No value supplied for os_requirements, setting default value of: 
JSSImporter: No value supplied for package_notes, setting default value of: 
JSSImporter: No value supplied for package_reboot, setting default value of: False
JSSImporter: No value supplied for JSS_SUPPRESS_WARNINGS, setting default value of: True
`

`		<key>SELF_SERVICE_DESCRIPTION</key>
		<string>You can see and find your files in Google Drive on your computer with Drive File Stream. You can save specific files and folders offline, including Team Drives; View and organize your files in Finder on macOS; Open files with apps installed on your computer; See all of your files in Google Drive without using storage space on your computer.</string>
		<key>SELF_SERVICE_ICON</key>
		<string>~/Library/AutoPkg/service-icons/GoogleDriveFilestream.png</string>`

And thank you for responding so quickly!
